pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveORCSerdeResolver.java

```diff
@@ -0,0 +1,150 @@
+            serdeType = HiveInputFormatFragmenter.PXF_HIVE_SERDES.ORC_SERDE;
+        } else {
+            throw new UnsupportedTypeException("Unsupported Hive Serde: " + serdeEnumStr);
+        }
+        parts = new StringBuilder();
+        partitionKeys = toks[HiveInputFormatFragmenter.TOK_KEYS];
+        parseDelimiterChar(input);
+        collectionDelim = input.getUserProperty("COLLECTION_DELIM") == null ? COLLECTION_DELIM
+                : input.getUserProperty("COLLECTION_DELIM");
+        mapkeyDelim = input.getUserProperty("MAPKEY_DELIM") == null ? MAPKEY_DELIM
+                : input.getUserProperty("MAPKEY_DELIM");
+    }
+
+    @Override
+    void initPartitionFields() {
+        numberOfPartitions = initPartitionFields(parts);
+    }
+
+    /**
+     * getFields returns a singleton list of OneField item.
+     * OneField item contains two fields: an integer representing the VARCHAR type and a Java
+     * Object representing the field value.
+     */
+    @Override
+    public List<OneField> getFields(OneRow onerow) throws Exception {
+
+        Object tuple = deserializer.deserialize((Writable) onerow.getData());
+        // Each Hive record is a Struct
+        StructObjectInspector soi = (StructObjectInspector) deserializer.getObjectInspector();
+        List<OneField> record = traverseStruct(tuple, soi, false);
+
+        return record;
+
+    }
+
+    /*
+     * Get and init the deserializer for the records of this Hive data fragment.
+     * Suppress Warnings added because deserializer.initialize is an abstract function that is deprecated
+     * but its implementations (ColumnarSerDe, LazyBinaryColumnarSerDe) still use the deprecated interface.
+     */
+    @SuppressWarnings("deprecation")
+	@Override
+    void initSerde(InputData input) throws Exception {
+        Properties serdeProperties = new Properties();
+        int numberOfDataColumns = input.getColumns() - numberOfPartitions;
+
+        LOG.debug("Serde number of columns is " + numberOfDataColumns);
+
+        StringBuilder columnNames = new StringBuilder(numberOfDataColumns * 2); // column + delimiter
+        StringBuilder columnTypes = new StringBuilder(numberOfDataColumns * 2); // column + delimiter
+        String delim = "";
+        for (int i = 0; i < numberOfDataColumns; i++) {
+            ColumnDescriptor column = input.getColumn(i);
+            String columnName = column.columnName();
+            String columnType = HiveInputFormatFragmenter.toHiveType(DataType.get(column.columnTypeCode()), columnName);
+            columnNames.append(delim).append(columnName);
+            columnTypes.append(delim).append(columnType);
+            delim = ",";
+        }
+        serdeProperties.put(serdeConstants.LIST_COLUMNS, columnNames.toString());
+        serdeProperties.put(serdeConstants.LIST_COLUMN_TYPES, columnTypes.toString());
+
+        if (serdeType == HiveInputFormatFragmenter.PXF_HIVE_SERDES.ORC_SERDE) {
```

Comment:
Why do we need check this if we are explicitly assigning certain value to this private filed before?